Package Updates + Instructions in README

iamanigeeit opened the pull request:
Upgrade node packages, fix non-compile on `state-machine-abc.md` and `docusaurus.config.js`, use `npm ci` instead of `npm install`.

Updated README.md.

Clean up Week 1.

kurniawano reviewed (changes requested):
@iamanigeeit , Can you leave the original getting started and add your notes as a separate section if we want to use your way. The original way is still correct and can work. So create a new sub heading in the README for alternative way.

iamanigeeit commented:
Hi @kurniawano the reason i changed the packages was due to npm install issues from having old packages (see below). If you are ok with it, i will see how to create a separate branch for different configs.

```sh
$npm install
npm warn deprecated stable@0.1.8: Modern JS already guarantees Array#sort() is a stable sort, so this library is deprecated. See the compatibility table on MDN: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/sort#browser_compatibility
npm warn deprecated source-map-url@0.4.1: See https://github.com/lydell/source-map-url#deprecated
npm warn deprecated urix@0.1.0: Please see https://github.com/lydell/urix#deprecated
npm warn deprecated trim@0.0.1: Use String.prototype.trim() instead
npm warn deprecated resolve-url@0.2.1: https://github.com/lydell/resolve-url#deprecated
npm warn deprecated source-map-resolve@0.5.3: See https://github.com/lydell/source-map-resolve#deprecated
npm warn deprecated w3c-hr-time@1.0.2: Use your platform's native performance.now() and performance.timeOrigin.
npm warn deprecated chokidar@2.1.8: Chokidar 2 does not receive security updates since 2019. Upgrade to chokidar 3 with 15x fewer dependencies
npm warn deprecated querystring@0.2.0: The querystring API is considered Legacy. new code should use the URLSearchParams API instead.
npm warn deprecated uuid@3.4.0: Please upgrade  to version 7 or higher.  Older versions may use Math.random() in certain circumstances, which is known to be problematic.  See https://v8.dev/blog/math-random for details.
npm warn deprecated svgo@1.3.2: This SVGO version is no longer supported. Upgrade to v2.x.x.

added 1721 packages, and audited 1722 packages in 54s

291 packages are looking for funding
  run `npm fund` for details

79 vulnerabilities (3 low, 20 moderate, 51 high, 5 critical)

To address issues that do not require attention, run:
  npm audit fix

To address all issues (including breaking changes), run:
  npm audit fix --force

Run `npm audit` for details.
```

kurniawano commented:
@iamanigeeit , the request for change is not on the npm packages but on the README markdown. Currently, your pull request remove the original instructions. 

Can you keep the original instructions?
